src/modules/hardwareWallet/hooks/useHwListener.js

```diff
@@ -0,0 +1,28 @@
+import { useDispatch } from 'react-redux';
+import { useEffect } from 'react';
+import { IPC_MESSAGES } from '@libs/hwServer/constants';
+import {
+  setDeviceListChanged,
+  setDeviceUpdated,
+} from 'src/modules/hardwareWallet/store/hardwareWalletActions';
+
+const { DEVICE_LIST_CHANGED, DEVICE_UPDATE } = IPC_MESSAGES;
+
+function useHwListener() {
+  const dispatch = useDispatch();
+
+  const { ipc } = window;
+
+  if (!ipc) return;
+
+  useEffect(() => {
+    ipc.on(DEVICE_LIST_CHANGED, (action, data) => {
```

Comment:
please use hwManager instade of direct ipc call

```suggestion
hwManager.subscribe(DEVICE_LIST_CHANGED, (action, data) => {
```

Reply:
@soroushm we dont have that method on the hwManager. Do you mean that i should use:
 `HwClient.subscribe(`

or should i create that function on the HwManager?

Reply:
hwManager extended hwClient so  subscirbe is included in hwmanager 
HWManager is a singlton class witch hwClient is not so we need to use hwManager

Reply:
Oh, yes that is true, will change so we are consistent on how we talk to the ledger